lib/compiler-llvm/src/compiler.rs

```diff
@@ -194,6 +198,12 @@ impl LLVMCompiler {
         let memory_buffer = target_machine
             .write_to_memory_buffer(&merged_module, FileType::Object)
             .unwrap();
+        if let Some(ref callbacks) = self.config.callbacks {
+            callbacks.obj_memory_buffer(
+                &CompiledFunctionKind::Local(function_body_inputs.iter().next().unwrap().0),
```

Comment:
As a reader of this code with no context, it's not obvious to me why this is correct, a comment about why the `unwrap` won't fail would be useful here (why do we know that there is at least one element?)

Reply:
Excellent catch! Really, why am I passing in one random function here. No, we should have an enum entry that correctly describes what we're actually calling the callback for.

Done.